### ISSUE TITLE
0.12 Roadmap RFC

### ISSUE THREAD

**Issue by sgrif** [diesel-rs/diesel#761]
This is the current roadmap for 0.12, which is targeted for *probably* late March (👶 makes concrete release schedules a bit harder). If there is a specific issue or feature which you need or would like to see, please comment here. Though do keep in mind that doesn't guarantee that it will be prioritized for 0.12.

Targeted features:

- Support the majority of PG upsert (#196)
- Operator support for PG JSON
- Datetime/Timestamp support for MySQL
- Decision on #747 
- Allow `infer_schema!` to support types which are not in `diesel::types`
- 3+ table joins
    - I'm exploring some ideas that don't require specialization, but they may not pan out. No promises on this one.

Not specifically targeted, but likely to be implemented:

- Support for PG network types
- Support for PG money

Non-goals for 0.12

- Async IO
    - We have a 90% implemented driver, but the API is not one that we're happy with. Upstream changes need to be made to tokio/futures before we can progress further on this feature
- Additional backends
    - Unless there is an overwhelming number of requests, it's unlikely that we'll be adding support for additional backends in Diesel proper. The design is intended to allow for third party crates to add support for additional backends, and the core team is happy to offer support for anyone wishing to do so.

**Comment by euclio** [comment]
I would like to see #184 resolved. As I said on the issue, I'd like to help with this but I'm having trouble finding time.

**Comment by jasonl** [comment]
For me, the ability to join to more than one table would be amazing. I'd like to help, but a lot of the internals of diesel are way beyond my Rust ken.
